core/flow/tc_throttling_test.go

```diff
@@ -76,16 +76,17 @@ func TestThrottlingChecker_DoCheckQueueingParallel(t *testing.T) {
 
 	tc := NewThrottlingChecker(nil, uint32(timeoutMs), uint32(intervalMs))
 
-	assert.True(t, tc.DoCheck(nil, 1, threshold) == nil)
-
 	wg := &sync.WaitGroup{}
 	gc := 24
 	wg.Add(gc)
 
-	var waitCount, blockCount int32 = 0, 0
+	var passCount, waitCount, blockCount int32 = 0, 0, 0
 	for i := 0; i < gc; i++ {
 		go func() {
 			res := tc.DoCheck(nil, 1, threshold)
+			if res.IsPass() {
```

Comment:
Add a latch before doing check may be better, which can let the goroutine concurrency more reasonable.